### ISSUE TITLE
Apache Lucene Update Warnings are shown when using Java 25

### ISSUE THREAD

**Issue by http418teapot** [dependency-check/DependencyCheck#8029]
**Precondition**
- [ x] I checked the issues list for existing open or closed reports of the same problem.

**Describe the bug**
When running the dependency check maven plugin on a project build with Java 25, warnings are being logged by apache lucene that it should be updated and maybe will not run in future versions :

```
WARNING: A restricted method in java.lang.foreign.Linker has been called
WARNING: java.lang.foreign.Linker::downcallHandle has been called by org.apache.lucene.store.PosixNativeAccess in an unnamed module (file:/builds/.m2/repository/org/apache/lucene/lucene-core/9.12.0/lucene-core-9.12.0.jar)
WARNING: Use --enable-native-access=ALL-UNNAMED to avoid a warning for callers in this module
WARNING: Restricted methods will be blocked in a future release unless native access is enabled
Oct 12, 2025 11:20:28 AM org.apache.lucene.store.MemorySegmentIndexInputProvider <init>
INFO: Using MemorySegmentIndexInput and native madvise support with Java 21 or later; to disable start with -Dorg.apache.lucene.store.MMapDirectory.enableMemorySegments=false
```

**Version of dependency-check used**
The problem occurs using version 12.1.6 of the maven plugin

**Expected behavior**
No warnings are given by Apache Lucene when running Dependency Check on a project build with Java 25
